app/models/course/lesson_plan_item.rb

```diff
@@ -0,0 +1,21 @@
+class Course::LessonPlanItem < ActiveRecord::Base
+  actable
+  stampable
+  validates :base_exp, numericality: { only_integer: true }
+  validates :time_bonus_exp, numericality: { only_integer: true }
+  validates :extra_bonus_exp, numericality: { only_integer: true }
+
+  after_initialize :set_default_values
+
+  def total_exp
+    base_exp + time_bonus_exp + extra_bonus_exp
```

Comment:
Can exp be <code>nil</code> ? if anyone of them is nil, this function will break.

If they cannot be <code>nil</code>, you should enforce this at db level.